packages/libs/react-ui/src/components/Modal/Modal.tsx

```diff
@@ -1,68 +1,55 @@
 'use client';
-import { Card } from '@components/Card';
-import { SystemIcon } from '@components/Icon';
-import { Heading } from '@components/Typography/Heading/Heading';
-import FocusTrap from 'focus-trap-react';
-import type { FC } from 'react';
-import React from 'react';
-import {
-  background,
-  closeButton,
-  modal,
-  titleWrapper,
-  wrapper,
-} from './Modal.css';
-import { useModal } from './useModal';
 
-export interface IModalProps {
-  children: React.ReactNode;
-  title?: string;
-  onClose?: () => void;
+import { mergeRefs } from '@react-aria/utils';
+import type { FC, ReactElement, Ref } from 'react';
+import React, { cloneElement, useRef } from 'react';
+import type { AriaModalOverlayProps, ModalOverlayAria } from 'react-aria';
+import { Overlay, useModalOverlay } from 'react-aria';
+import type { OverlayTriggerState } from 'react-stately';
+import { underlayClass } from './Modal.css';
+
+export interface IModalProps extends AriaModalOverlayProps {
+  state: OverlayTriggerState;
+  children:
+    | ReactElement
+    | ((
+        modalProps: ModalOverlayAria['modalProps'],
+        ref: Ref<HTMLDivElement>,
+      ) => ReactElement);
 }
 
-export const Modal: FC<IModalProps> = ({ children, title, onClose }) => {
-  const { clearModal } = useModal();
+export const Modal: FC<IModalProps> = ({
+  children,
+  state,
+  isDismissable = true,
+  isKeyboardDismissDisabled,
+}) => {
+  const nodeRef = useRef<HTMLDivElement | null>(null);
+  const { modalProps, underlayProps } = useModalOverlay(
+    {
+      isDismissable,
+      isKeyboardDismissDisabled,
+    },
+    state,
+    nodeRef,
+  );
 
-  function handleCloseModal(): void {
-    onClose?.();
-    clearModal();
+  if (!state.isOpen) {
+    return null;
   }
 
   return (
-    <>
-      <FocusTrap
-        focusTrapOptions={{
-          fallbackFocus: '[data-cy="modal-background"]',
-        }}
-      >
-        <div>
-          <button
-            data-cy="modal-background"
-            className={background}
-            onClick={handleCloseModal}
-          />
-          <div className={wrapper} data-cy="modal" data-testid="kda-modal">
-            <section className={modal}>
-              <Card fullWidth>
-                <div className={titleWrapper}>
-                  <Heading as="h3">{title}</Heading>
-                </div>
-
-                <button
```

Comment:
the dialog now scrolls the complete content.
I think this is fine, but in designs discussed with isa (for the docs search), the scrolling could be controlled by the content it self.
as in example below the header + the search and tabs always stay visible.
Regardless I think the close button should always be visible and not scroll.
Can we discuss with Isa?

![image](https://github.com/kadena-community/kadena.js/assets/4015521/c1e98df0-26ac-4831-a446-fe27eb782c23)

Reply:
hmm yeah, I think it's good for us to have a chat about this

Reply:
I've updated it to have Header, Footer, and Content cubcomponents. The Content section will scroll if there is overflow